### PR TITLE
Site Selector: enter key select (updated)

### PR DESCRIPTION
**Pull request by adambbecker**
#### Updated

This is an updated version of #2646 which was reverted by #2938 because of a bug that would occur if you searched for a site, then clicked on it, it wouldn't navigate.

/cc @mtias, @umurkontaci

---

On the site selector, if search is utilized to filter the sites list and only one result is visible, the enter key will now select that site. Think of it like a quick switcher.

In the future, I'd like to expand this to account for other keyboard events like arrow keys, etc. but in order to keep this PR as focused as possible, it's **only focused** on the specific case of hitting the enter key when filtered to one site.

##### Example screencast

enterkey

## Testing

1. Open site picker
2. Enter search term (only visible if account has many sites)
3. When only one site is visible, hit **enter**
4. You should navigate to the appropriate url for that filtered site
5. **Bug check**
  * Search for a site to filter the list, then click on the site. Should still navigate as intended.

### CODE REVIEW

File: client/components/site-selector/index.jsx [hunk 1]
```diff
@@ -63,6 +62,20 @@ export default React.createClass( {
 		this.setState( { search: terms } );
 	},
 
+	onKeyDown( event ) {
+		var filteredSites;
```
Comment (gwwar) [inline client/components/site-selector/index.jsx]: `const`?

File: client/components/site-selector/index.jsx [hunk 2]
```diff
@@ -63,6 +62,20 @@ export default React.createClass( {
...
+
+		if ( event.keyCode === 13 ) {
+			// enter key
+			filteredSites = this.getFilteredSites();
+
+			if ( filteredSites.length === 1 && this.props.siteBasePath ) {
```
Comment (gwwar) [inline client/components/site-selector/index.jsx]: Maybe add `const site = filteredSites[ 0 ];` for readability.

**Comment by gwwar** [comment]
:+1: Tested on Chrome, FF, Safari, Edge and IE11. After a rebase this is good to :ship: 

Feel free to ping me for another look if the rebase is more involved.

**Comment by rickybanister** [comment]
This seems like it was ready to go before getting outdated, can we revive it?

**Comment by adambbecker** [comment]
@rickybanister: Thanks for the reminder, got stuck on the backlog.

**Comment by lancewillett** [comment]
Closing this one as no movement in several months — feel free to re-open if it's actionable and can be pushed in.

**Pull request closed without merging**